File: Examples/Blueprint Example (Obj-C)/Blueprint Example (Obj-C)/Pet.m
//
//  Pet.m
//  Blueprint Example (Obj-C)
//
//  Created by Hunter on 5/12/16.
//  Copyright © 2016 Blueprint Project. All rights reserved.
//

#import "Pet.h"

@implementation Pet

-(void)giveToy:(Toy *)toy
{
    NSMutableArray *toy_ids = @[].mutableCopy;
    
    if(self[@"toy_ids"] != nil) {
        toy_ids = [self[@"toy_ids"] mutableCopy];
    }
    
    [toy_ids addObject:toy.objectId];
    
    self[@"toy_ids"] = toy_ids;
}

@end

File: Examples/Blueprint Example/Models/Pet.swift
//
//  Pet.swift
//  Blueprint Example
//
//  Created by Hunter on 5/1/16.
//  Copyright © 2016 Blueprint Project. All rights reserved.
//

import Cocoa

class Pet: BPModel {

    func giveToy(_ toy: Toy) {
        var toy_ids = [String]();
        
        if self["toy_ids"] != nil {
            toy_ids = self["toy_ids"] as! [String]
        }
        
        toy_ids.append(toy.objectId)
        
        self["toy_ids"] = toy_ids
    }
    
}
